src/ScrumBoardItBundle/Services/Api/ApiVisitor.php

```diff
@@ -0,0 +1,295 @@
+<?php
+
+namespace ScrumBoardItBundle\Services\Api;
+
+use Symfony\Component\HttpFoundation\Request;
+use ScrumBoardItBundle\Form\Type\Search\VisitorSearchType;
+use ScrumBoardItBundle\Entity\Issue\Task;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class ApiVisitor extends AbstractApi
+{
+    /**
+     * @var array
+     */
+    private $issues;
+    
+    /**
+     * Issue id incrementor.
+     *
+     * @var number
+     */
+    private $currentIssueIndex = 0;
+
+    /**
+     * Array of printed issues' ids.
+     *
+     * @var array
+     */
+    private $printedIssues;
+
+    public function __construct(RequestStack $requestStack, $token, $config, $apiCaller)
+    {
+        $session = $requestStack->getCurrentRequest()->getSession();
+        parent::__construct($token, $config, $apiCaller);
+        $this->printedIssues = $session->get('printed_issues', []);
+        $this->generateIssues();
+    }
+
+    /**
+     * Generate an array of issues.
+     */
+    private function generateIssues()
+    {
+        $this->issues = array(
+            1 => array(
```

Comment:
this key is the number of the repository, it is necessary. However, i removed keys of issues, never used.